synapse/handlers/sso.py

```diff
@@ -810,6 +817,10 @@ def is_allowed_mime_type(content_type: str) -> bool:
                 str(avatar_mxc_url),
             )
 
+            # delete previous avatar, if one existed
+            if delete_previous_avatar and existing_media_id is not None:
+                await self._media_repo.delete_local_media_ids(existing_media_id)
```

Comment:
note that a user can typically set their `avatar_url` to whatever they like, including another user's piece of media or another user's avatar. Allowing them to delete a piece of media this way is therefore dangerous.
(It also means their previous messages will be left without an avatar, which might be unwanted depending on the circumstances.)

I'm not sure what the best way to approach this is; I suppose it will depend on the motivation here.